include/oneapi/tbb/detail/_flow_graph_item_buffer_impl.h

```diff
@@ -102,22 +109,42 @@ class item_buffer {
 #endif
     }
 
+#if __TBB_PREVIEW_FLOW_GRAPH_TRY_PUT_AND_WAIT
+    void set_my_item(size_t i, const item_type& o, message_metainfo&& metainfo) {
+        if(element(i).state != no_item) {
+            destroy_item(i);
+        }
+        new(&(element(i).item)) item_type(o);
+        new(&element(i).metainfo) message_metainfo(std::move(metainfo));
```

Comment:
We can skip increasing reference counter because we moving ownership?

Reply:
Yes, exactly. I have added a comment here.